script2.js

```diff
@@ -0,0 +1,94 @@
+const API = 'https://pokeapi.co/api/v2/pokemon/'
+const wrapper = document.querySelector('wrapper')
+const main = document.querySelector('main')
+const pokemonOne = document.querySelector('.pokemonOne')
+const backCards = Array.from(document.querySelectorAll('.backCard'))
+const next = document.querySelector('.next')
+const previous = document.querySelector('.previous')
+const page = document.querySelector('.currentPage');
+let localDataBase = ['empty'];
+let pokemonsFetchesSoFar = 0;
+let currentPage = 0;
+let c = pokemonsFetchesSoFar; // counter
+let allPages = [];
+
+async function fetchPokemons() {
+    let pokemonsFetched = [];
+
+    for (let i = 0; i < 3; i++) {
+        c++
+        pokemonsFetched.push(await (await fetch(API + c)).json());
+    }
+    localDataBase.push(pokemonsFetched);
+    return pokemonsFetched
+}
+
+async function domUpdate(fn, direction) {
+
+    if (direction === 'next') {
+        currentPage++;
+    } else if (direction === 'previous' && currentPage > 1) {
+        currentPage--;
+    } else {
+        return;
+    }
+
+    if (currentPage >= localDataBase.length) {
+        await fn();
+    }
+
+    pokemonOne.innerHTML = ` <img src="${localDataBase[currentPage][1].sprites.other.dream_world.front_default}"
+    alt="${localDataBase[currentPage][1].name}">
+    <div class="imgOverlay"></div>
+    <div class="shineEffect"></div>
+    <div class="name">${localDataBase[currentPage][1].name.toUpperCase()}</div>
+    <div class="info">${localDataBase[currentPage][1].name.toUpperCase()}</div>`;
+    backCards[0].innerHTML = `<img src="${localDataBase[currentPage][0].sprites.other.dream_world.front_default}"><div class="shineEffect"></div><div class="imgOverlay"></div><div class="name">${localDataBase[currentPage][0].name.toUpperCase()}</div>`;
+    backCards[1].innerHTML = `<img src="${localDataBase[currentPage][2].sprites.other.dream_world.front_default}"><div class="shineEffect"></div><div class="imgOverlay"></div><div class="name">${localDataBase[currentPage][2].name.toUpperCase()}</div>`;
+
+    nav = () => {
+        const pagesHTML = localDataBase.toSpliced(0, 1).map((_, i) => `<span class="num${(i+1)}">${i + 1}</span>`)
+        page.innerHTML = `<p>${currentPage}</p><div class='restNav'><p>${pagesHTML}</p></div`;
+        allPages = document.querySelectorAll('p span')
+        allPages.forEach((p, i) => {
+            p.addEventListener('click', () => {
+                console.log(i);
+            })
+        })
+    }
```

Comment:
sparametryzować funkcję i wyciągnąć ją z `domUpdate`
```suggestion
    nav = () => {
        const pagesHTML = localDataBase.toSpliced(0, 1).map((_, i) => `<span class="num${(i+1)}">${i + 1}</span>`)
        page.innerHTML = `<p>${currentPage}</p><div class='restNav'><p>${pagesHTML}</p></div`;
        allPages = document.querySelectorAll('p span')
        allPages.forEach((p, i) => {
            p.addEventListener('click', () => {
                console.log(i);
            })
        })
    }
```